src/pages/index.js

```diff
@@ -4,57 +4,94 @@ import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 import UserInfo from "../components/UserInfo.js";
 /* DECLARATIVE SECTION */
 import {
-  initialCards,
   configuration,
-  profile,
   buttonEditProfile,
   profileName,
   profileDescription,
+  buttonUpdateAvatar,
+  profileAvatar,
   modalWindowProfile,
-  formProfile,
   inputProfileName,
   inputProfileDescription,
   buttonAddCard,
   modalWindowCardAdd,
   formCardAdd,
-  inputCardTitle,
-  inputCardLink,
   modalWindowCardView,
   modalImage,
   modalCaption,
+  modalWindowCardDelete,
+  modalWindowProfileAvatar,
+  inputProfileAvatar,
   cardsGallery,
+  submitProfileInfo,
+  submitProfileAvatar,
+  submitCardAdd,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    "content-type": "application/json",
+    authorization: "c48564f3-ecfb-4a65-b35d-031aabc842ce",
+  },
+});
+
+let cardSection; // since the card section is being tied to the DOM and is needed to both init and add cards later on, it would need to be declared at the global scope and can be reassigned with let.
+let userId;
+
+api
+  .getUserAndCards([
+    // this is a Promise array that would get fulfilled when all subsequent promises are fulfilled too (&&-style)
```

Comment:
`Promise.all` should get the promisses which haven't used their `then` blocks yet

So, it should be like this:



```js

api.getUserAndCards([api.getUserProfileData(), api.getInitialCards()])
// destructure the response
  .then(([{ _id, name, about, avatar } , cardArray]) => {
      // set all the data
  })
  .catch(err => {
    // catch possible errors
  });

```